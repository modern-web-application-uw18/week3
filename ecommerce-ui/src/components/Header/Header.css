.Header {
    background-color: #000000;
    width: 100%;
    height: 65px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 40px;
    color: #ffffff;
    margin-bottom: 0;
    padding-right: 10px;
    padding-left: 10px;
    display: grid;
    justify-content: space-between;
    grid-template-columns: 50% 50%;
    grid-auto-rows: 50%;
    grid-gap: 5px;


}


@media (max-width: 650px) {
    .Header {
        font-size: 30px;
        padding-top: 10px;
    }
}