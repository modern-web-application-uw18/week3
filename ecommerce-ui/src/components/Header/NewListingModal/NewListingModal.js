import React, { Component } from 'react';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';
import Form from 'react-bootstrap/lib/Form';
import FormGroup from 'react-bootstrap/lib/FormGroup';
import Col from 'react-bootstrap/lib/Col';
import FormControl from 'react-bootstrap/lib/FormControl';
import PropTypes from 'prop-types';


class NewListingModal extends Component {


    state = {
        title: '',
        houseType: '',
        image: '',
        city: '',
        country: '',
        cost: '',
        name: ''
    }


    handleNameChange = (e) => {
        let name = e.target.value;
        this.setState((prevState, props) => {
            return { name: name };
        });
    }

    handleCityChange = (e) => {
        let city = e.target.value;
        this.setState((prevState, props) => {
            return { city: city };
        });
    }

    handleCountryChange = (e) => {
        let country = e.target.value;
        this.setState((prevState, props) => {
            return { country: country };
        });
    }

    handleTitleChange = (e) => {
        let title = e.target.value;
        this.setState((prevState, props) => {
            return { title: title };
        });
    }

    handleHouseTypeChange = (e) => {
        let houseType = e.target.value;
        this.setState((prevState, props) => {
            return { houseType: houseType };
        });
    }

    handleImageURLChange = (e) => {
        let image = e.target.value;
        this.setState((prevState, props) => {
            return { image: image };
        });
    }

    handleCostChange = (e) => {
        let cost = e.target.value;
        this.setState((prevState, props) => {
            return { cost: cost };
        });
    }


    submitForm = () => {

        let proceed = true;

        this.props.mockDataArr.filter(
            (obj, idx) => {
                if (obj.title === this.state.title) {
                    proceed = false;
                    alert('Title already exists. Please choose another one!');
                    this.setState((prevState, props) => {
                        return { title: '' };
                    });
                }
                return null;
            }
        );

        if (this.getValidationState() === 'success' && proceed) {

            this.props.dataHandler(this.state);
            this.props.handleFormClose();
            this.setState((prev, props) => {
                return {
                    title: '',
                    houseType: '',
                    image: '',
                    city: '',
                    country: '',
                    cost: '',
                    name: ''
                }
            });
        }

    }

    getValidationState = () => {


        const length = this.state.title.length;
        if (length > 0) return 'success';
        else if (length > 0) return 'warning';
        else if (length === 0) return 'error';
        return null;
    }

    render() {
        return (
            <Modal show={this.props.formShow} onHide={this.props.handleFormClose}>
                <Modal.Header closeButton>
                    <Modal.Title>Add a New Listing</Modal.Title>
                </Modal.Header>
                <Modal.Body>

                    <Form horizontal>

                        <FormGroup controlId="formTitle" validationState={this.getValidationState()}>
                            <Col sm={3}>Title</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.title} onChange={this.handleTitleChange} placeholder="Title" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formName">
                            <Col sm={3}>Name</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.name} onChange={this.handleNameChange} placeholder="Name" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formCity">
                            <Col sm={3}>City</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.city} onChange={this.handleCityChange} placeholder="City" />
                            </Col>
                        </FormGroup>


                        <FormGroup controlId="formCountry">
                            <Col sm={3}>Country</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.country} onChange={this.handleCountryChange} placeholder="Country" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formHouseType">
                            <Col sm={3}>House Type</Col>
                            <Col sm={9}>
                                <FormControl componentClass="select" value={this.state.houseType} onChange={this.handleHouseTypeChange} placeholder="Select">
                                    <option value="">...</option>
                                    <option value="Apartment">Apartment</option>
                                    <option value="Entire House">Entire House</option>
                                </FormControl>
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formImage">
                            <Col sm={3}>Image URL</Col>
                            <Col sm={9}>
                                <FormControl type="text" value={this.state.imageURL} onChange={this.handleImageURLChange} placeholder="Image URL" />
                            </Col>
                        </FormGroup>

                        <FormGroup controlId="formCost">
                            <Col sm={3}>Cost</Col>
                            <Col sm={9}>
                                <FormControl type="number" value={this.state.cost} onChange={this.handleCostChange} placeholder="Cost" />
                            </Col>
                        </FormGroup>

                        <FormGroup>
                            <Col smOffset={3} sm={9}>
                                <Button className="submitBtn" onClick={this.submitForm}>Submit New Listing</Button>
                            </Col>
                        </FormGroup>

                    </Form>


                </Modal.Body>
                <Modal.Footer>
                    <Button onClick={this.props.handleFormClose}>Close</Button>

                </Modal.Footer>
            </Modal>
        )
    }


}


NewListingModal.propTypes = {
    formShow: PropTypes.bool.isRequired,
    handleFormClose: PropTypes.func.isRequired,
    dataHandler: PropTypes.func.isRequired,
    mockDataArr: PropTypes.array.isRequired
}

export default NewListingModal;