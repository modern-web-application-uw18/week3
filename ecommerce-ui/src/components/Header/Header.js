import React, { Component } from 'react';
import classes from './Header.css';
import Cart from './Cart/Cart';
import NewListingModal from './NewListingModal/NewListingModal';

import MdShoppingCart from 'react-icons/lib/md/shopping-cart';
import MdAddBox from 'react-icons/lib/md/add-box';

class Header extends Component {
  state = {
    cartShow: false,
    formShow: false
  }

  handleCartClose = () => {
    this.setState({ cartShow: false });
  }

  handleCartShow = () => {
    this.setState({ cartShow: true });
  }


  handleFormClose  = () => {
    this.setState({ formShow: false });
  }

  handleFormShow  = () => {
    this.setState({ formShow: true });
  }

  render() {

    return (
      <div className={classes.Header}>

        <div>
          <button type="button" className="btn btn-danger" style={{ width: '150px', height: '40px', border: '0px' }} onClick={this.handleFormShow}>
            Add Listing &nbsp;&nbsp; <MdAddBox size={30} />
          </button>
        </div>

        <div style={{ textAlign: 'right' }}>
          <button type="button" className="btn btn-default" style={{ width: '150px', height: '40px', border: '0px' }} onClick={this.handleCartShow}>
            Checkout &nbsp;&nbsp; <MdShoppingCart size={30} />
          </button>
        </div>


        <Cart cartShow={this.state.cartShow}  handleCartClose={this.handleCartClose} cartContent={this.props.cartContent} removeListingFromTheCartHandler={this.props.removeListingFromTheCartHandler} />
        <NewListingModal formShow={this.state.formShow} handleFormClose={this.handleFormClose} dataHandler={this.props.dataHandler} mockDataArr={this.props.mockDataArr}/>


      </div>
    )
  }
}


export default Header;
