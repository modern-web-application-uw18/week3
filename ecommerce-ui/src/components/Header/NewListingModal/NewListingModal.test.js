import React from 'react';
import { shallow, mount} from 'enzyme';
import NewListingModal from './NewListingModal';
import Header from '../../Header/Header';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';
import Form from 'react-bootstrap/lib/Form';
import FormGroup from 'react-bootstrap/lib/FormGroup';
import Col from 'react-bootstrap/lib/Col';
import FormControl from 'react-bootstrap/lib/FormControl';


describe('<NewListingModal />', () => {

    let mockListing = {
        "title": "Centrally-located Manhattan studio 2",
        "houseType": "Apartment",
        "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
        "location": {
            "city": "New York",
            "country": "USA"
        },
        "payment": {
            "cost": 105,
            "description": "Free cancellation"
        },
        "host": {
            "name": "Robert",
            "isSuperhost": true
        },
        "rating": {
            "stars": 3,
            "reviews": 133
        }
    }

    it('Snapshot test', () => {
        const wrapper2 = mount(<Header cartContent={[mockListing]} />);
        let fn = wrapper2.instance().handleFormClose;


        const wrapper = mount(<NewListingModal handleFormClose={fn}/>);

        // wrapper.instance().mockDataHandler(mockListing);
        // wrapper.instance().removeListingFromTheCartHandler(mockListing);
        // wrapper.instance().addListingToTheCartHandler(mockListing);

        expect(wrapper).toMatchSnapshot();
    })
})