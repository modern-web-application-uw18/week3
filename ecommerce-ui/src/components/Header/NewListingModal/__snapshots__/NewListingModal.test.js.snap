// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewListingModal /> Snapshot test 1`] = `
<NewListingModal
  handleFormClose={[Function]}
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropClassName="modal-backdrop"
      backdropTransition={[Function]}
      containerClassName="modal-open"
      enforceFocus={true}
      keyboard={true}
      manager={
        ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        }
      }
      onEntering={[Function]}
      onExited={[Function]}
      onHide={[Function]}
      renderBackdrop={[Function]}
      restoreFocus={true}
      show={false}
      transition={[Function]}
    />
  </Modal>
</NewListingModal>
`;
